### ISSUE TITLE
Empty String in Text Component causes error displayName of undefined

### ISSUE THREAD

**Issue by unuigbee** [facebook/react-native#10090]
### Issue Description

If a Text Component with a empty string gets rendered I receive an error saying: Unhandled JS Exception: Cannot read property 'displayName' of undefined_ 

### Steps to Reproduce / Code Snippets
Simply add a empty String within a Text component or a property whose value is an empty string

```
class WeatherProject extends Component {
  constructor(props) {
    super(props);
    this.state = {
      zipText: ''
    };
  }
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          You Input {this.state.zipText} Or this {''} errors out
        </Text>
      </View>
    );
  }
}
```
![image](https://cloud.githubusercontent.com/assets/6902308/18809929/0678912c-8281-11e6-8ad9-ecc830fef3af.png)

#### Expected Results
![image](https://cloud.githubusercontent.com/assets/6902308/18809905/5ec28d20-8280-11e6-8773-456e3316156b.png)

### Additional Information

* React Native version:0.34
* Platform(s) (iOS, Android, or both?): Both (iOS 10) and (Android 7.0 "Nougat")
* Operating System (macOS, Linux, or Windows?):MacOS (El Capitan 10.11.6)

**Comment by reneweb** [comment]
I tried to reproduce using your example, but it worked fine for me. I tested on iOS 10 and Android 7.0 emulator, both 0.34 and master branch and I am also running El Capitan 10.11.6.

Have you tried a fresh project with just the example you have posted here?

<img width="373" alt="react-native-empty-string-ios-10" src="https://cloud.githubusercontent.com/assets/6517897/18907848/3672b678-8565-11e6-96b9-00a87784880c.png">
